### PR TITLE
Mehran Graphs

### PR DESCRIPTION
**Pull request by tbhallett**
This is the definitive set of changes that enable production of the "Mehran Graphs"

* Builds on #54 
* ... which itself built on #46 

Before merging, I recommend, following the comments raised on an earlier PR that requests additional comments: [here ](https://github.com/globalfund-pri/model_code_pipeline/pull/54#discussion_r1875006942)and [here](https://github.com/globalfund-pri/model_code_pipeline/pull/54#discussion_r1874996632).

### CODE REVIEW

File: src/scripts/ic8/hiv/hiv_filehandlers.py [hunk 1]
```diff
@@ -728,33 +728,27 @@ def _turn_workbook_into_df(self, file: Path) -> pd.DataFrame:
             # Extract Step4 values for 2022
             step4_2022 = csv_df[(csv_df['scenario_descriptor'] == 'Step4') & (csv_df['year'] == 2022)]
 
-            # Merge Step4 values with Step13 rows
-            df_step13_updated = csv_df[mask_step13_2022].merge(
-                step4_2022[['country'] + column_names],
-                on='country',
-                suffixes=('', '_step4')
+            # Create a mapping for Step4 values (x, y, z) by country
```
Comment (tbhallett) [inline src/scripts/ic8/hiv/hiv_filehandlers.py:731]: explain

File: src/scripts/ic8/hiv/hiv_analysis.py [hunk 2]
```diff
@@ -142,3 +142,22 @@ def get_hiv_analysis(
         header=False
     )
 
+    list_of_dfs = list()  # list of mini dataframes for each indicator for each country
```
Comment (tbhallett) [inline src/scripts/ic8/hiv/hiv_analysis.py:145]: explain / remove?

**Comment by tbhallett** [comment]
- [x] Produces IC8 Analysis
- [x] Test-ic7 passes

File: src/scripts/ic8/hiv/hiv_filehandlers.py (hunk 1 above)
Comment (MikaelaSmitGlobalFund) [reply src/scripts/ic8/hiv/hiv_filehandlers.py:731]: @tbhallett , I have explained that in the Peter S branch

**Review by MikaelaSmitGlobalFund — APPROVED** [review]
@tbhallett I ran this and it works on my side now also, graphs look good. We can merge

**Review by MikaelaSmitGlobalFund — APPROVED** [review]

**Pull request merged**